libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java

```diff
@@ -235,10 +235,12 @@ public void clearMarkers() {
    *
    * @param location to update the icon and query the map
    */
-  public void updateLocation(Location location) {
-    List<Location> locations = new ArrayList<>(1);
-    locations.add(location);
-    updateLocation(locations);
+  public void updateLocation(@NonNull Location location) {
+    if (location != null) {
```

Comment:
sorry, `@NonNull` but we check it on null. should be `@Nullable`? 😄

Reply:
or don't check if is `null`

Reply:
yep, it's confusing to check `@NonNull` nullability
btw, I get compilation errors with calling that method. Have you checked that it compiles?